handlers/handlers.go

```diff
@@ -94,41 +79,73 @@ func PostTokenHandler(w http.ResponseWriter, r *http.Request) {
 	// /* Uncomment this for token generation to work w/o karen
 	if err = json.NewDecoder(resp.Body).Decode(&userBody); err != nil {
 		log.Printf(`Failed to authorize user, unable to parse response body of "%s" request: %s\n`, authRoute, err.Error())
-		http.Error(w, `Failed to authorize user`, http.StatusInternalServerError)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
 	// */ // Uncomment this for token generation to work w/o karen
 
-	token, err := createToken(userBody)
+	token, err := e.createToken(userBody)
 	if err != nil {
 		log.Println(`Failed to create token: `, err)
-		http.Error(w, `Failed to create token for authorized user`, http.StatusInternalServerError)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
 
 	json.NewEncoder(w).Encode(map[string]string{"token": token})
 }
 
-func forwardRequest(w http.ResponseWriter, r *http.Request) {
+func (e *Env) forwardRequest(w http.ResponseWriter, r *http.Request) {
 	r.RequestURI = ""
-	if resp, err := rc.Do(r); err != nil {
+	urlString := r.URL.String()
+	re := regexp.MustCompile("[^/]+")
+
+	// Parse the URL for the service name
+	appName := re.FindString(urlString)
+	if appName == "" {
+		log.Printf("Service name not provided")
+		http.Error(w, "Not Found", http.StatusNotFound)
+		return
+	}
+
+	// Check if app IP address is alreaday cached
```

Comment:
```suggestion
	// Check if app IP address is already cached
```

Reply:
I like to throw a little personality into my comments